src/lib/core.c

```diff
@@ -1127,6 +1187,33 @@ gpiod_chip_iter_failed_chip(struct gpiod_chip_iter *iter)
 	return iter->failed_chip;
 }
 
+struct gpiod_chip_bulk_iter *
+gpiod_chip_bulk_iter_new(struct gpiod_chip_bulk *bulk)
+{
+	struct gpiod_chip_bulk_iter *new;
+
+	new = zalloc(sizeof(*new));
+	if (!new)
+		return NULL;
+
+	new->bulk = bulk;
+
+	return new;
+}
+
+struct gpiod_chip *
+gpiod_chip_bulk_iter_next(struct gpiod_chip_bulk_iter *iter)
+{
+	if (iter) {
```

Comment:
This was necessary due to a possible failure of `zalloc` in the _iter_new called in the foreach macro.